modules/core/src/main/java/org/apache/ignite/internal/direct/stream/v2/DirectByteBufferStreamImplV2.java

```diff
@@ -1803,7 +1804,42 @@ protected Object read(MessageCollectionItemType type, MessageReader reader) {
 
     /** {@inheritDoc} */
     @Override public String toString() {
-        return S.toString(DirectByteBufferStreamImplV2.class, this);
+        return S.toString(DirectByteBufferStreamImplV2.class, this,
+            "colTypes", col == null ? null : lastTypesWithLimit(col),
+            "objArrTypes", objArr == null ? null : lastTypesWithLimit(Arrays.asList(objArr))
+        );
+    }
+
+    /**
+     * Builds a string showing the simple types of the elements of the list; if the list is longer than the
+     * limit, then only its final elements are considered and ellipsis represents the omitted part of the list.
+     *
+     * @param list List to represent.
+     * @return String like [Integer, String] if there are exactly 2 elements of types Integer and String, respectively.
+     */
+    private String lastTypesWithLimit(List<Object> list) {
```

Comment:
Do I understand correctly that this method can be static, and we can specifically test it?

I propose specifically to test this method better and make the limit an argument of the method, as an option, move this method to an auxiliary class.

Reply:
Made the method static and package local to make it easier to test it. I don't think it should be extracted to a common tool yet as it's only used in one class for now. If we need it elsewhere, we'll do it.